pipeline_manager/src/main.rs

```diff
@@ -475,12 +608,21 @@ struct CompileProjectRequest {
 ///
 /// The client should poll the `/project_status` endpoint
 /// for compilation results.
-///
-/// # HTTP errors
-///
-/// * `NOT_FOUND` - `project_id` does not exist in the database.
-/// * `CONFLICT` - project version specified in the request doesn't match the
-///   latest project version in the database.
+#[utoipa::path(
+    request_body = CompileProjectRequest,
+    responses(
+        (status = ACCEPTED, description = "Compilation request submitted."),
+        (status = NOT_FOUND
+            , description = "Specified `project_id` does not exist in the database."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Unknown project id '42'"))),
+        (status = CONFLICT
+            , description = "Project version specified in the request doesn't match the latest project version in the database."
+            , body = ErrorResponse
+            , example = json!(ErrorResponse::new("Outdated project version '{version}'"))),
+    ),
+    tag = "Project"
+)]
 #[post("/compile_project")]
```

Comment:
`POST /projects/{project_id}/compile`

Reply:
Isn't it better to do `POST /projects/compile` and extract `project_id` from JSON?  Otherwise, we have a mix of url-encoded and JSON arguments.

Reply:
Yes, I made a similar suggestion to that effect in another comment now that I understand the workflow better.